clients/src/main/java/org/apache/kafka/common/telemetry/internals/SinglePointMetric.java

```diff
@@ -16,22 +16,135 @@
+    }
+
+    public static SinglePointMetric create(MetricKey metricKey, Metric.Builder metric) {
+        return new SinglePointMetric(metricKey, metric);
+    }
+
+    /*
+        Methods to construct gauge metric type.
+     */
+    public static SinglePointMetric gauge(MetricKey metricKey, Number value, Instant timestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        return gauge(metricKey, point);
+    }
+
+    public static SinglePointMetric gauge(MetricKey metricKey, double value, Instant timestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        return gauge(metricKey, point);
+    }
+
+    /*
+        Methods to construct sum metric type.
+     */
+
+    public static SinglePointMetric sum(MetricKey metricKey, double value, boolean monotonic, Instant timestamp) {
+        return sum(metricKey, value, monotonic, timestamp, null);
+    }
+
+    public static SinglePointMetric sum(MetricKey metricKey, double value, boolean monotonic, Instant timestamp,
+        Instant startTimestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value);
+        if (startTimestamp != null) {
+            point.setStartTimeUnixNano(toTimeUnixNanos(startTimestamp));
+        }
+
+        return sum(metricKey, AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE, monotonic, point);
+    }
+
+    public static SinglePointMetric deltaSum(MetricKey metricKey, double value, boolean monotonic,
+        Instant timestamp, Instant startTimestamp) {
+        NumberDataPoint.Builder point = point(timestamp, value)
+            .setStartTimeUnixNano(toTimeUnixNanos(startTimestamp));
+
+        return sum(metricKey, AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA, monotonic, point);
+    }
+
+    /*
+        Helper methods to support metric construction.
+     */
+    private static SinglePointMetric sum(MetricKey metricKey, AggregationTemporality aggregationTemporality,
+        boolean monotonic, NumberDataPoint.Builder point) {
+        point.addAllAttributes(asAttributes(metricKey.tags()));
+
+        Metric.Builder metric = Metric.newBuilder().setName(metricKey.name());
+        metric
+            .getSumBuilder()
+            .setAggregationTemporality(aggregationTemporality)
+            .setIsMonotonic(monotonic)
+            .addDataPoints(point);
+        return create(metricKey, metric);
+    }
+
+    private static SinglePointMetric gauge(MetricKey metricKey, NumberDataPoint.Builder point) {
+        point.addAllAttributes(asAttributes(metricKey.tags()));
+
+        Metric.Builder metric = Metric.newBuilder().setName(metricKey.name());
+        metric.getGaugeBuilder().addDataPoints(point);
+        return create(metricKey, metric);
+    }
+
+    private static NumberDataPoint.Builder point(Instant timestamp, Number value) {
+        if (value instanceof Long || value instanceof Integer) {
+            return point(timestamp, value.longValue());
+        } else {
```

Comment:
else is not needed

Reply:
why not? This covers the case where a gauge may be of type Double / Float and ensures we preserve the underlying precision instead of truncating the values.

Reply:
oh I see, you mean just the else, sure that's fair

Reply:
Done.